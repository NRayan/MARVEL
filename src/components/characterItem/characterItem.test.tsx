// import { render } from "@testing-library/react-native";
import { render } from "@testing-library/react-native";
import React from "react";
import { mockCharacters, RenderWithNavigation, RenderWithTheme } from "../../tests";
import { CharacterItem } from "./";
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

const characters = mockCharacters();
const CharacterRender = () => (<CharacterItem character={characters[0]} />);

describe("characterItem", () => {

	it("Should Render correctly", () => {

		const tree = render(
			<RenderWithTheme>
				<RenderWithNavigation component={CharacterRender} />
			</RenderWithTheme>
		).toJSON();

		expect(tree).toMatchSnapshot();

	});
});