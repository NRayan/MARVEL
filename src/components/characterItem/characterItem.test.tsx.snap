// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`characterItem Should Render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="MockedScreen"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#21263e",
                "borderRadius": 10,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 150,
                "marginBottom": 10,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 10,
                "overflow": "hidden",
                "width": "100%",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "asd",
              }
            }
            style={
              Array [
                Object {
                  "aspectRatio": 0.8,
                  "height": "100%",
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFF",
                    "fontFamily": "Inter_700Bold",
                  },
                ]
              }
            >
              asd
            </Text>
            <Text
              ellipsizeMode="tail"
              numberOfLines={3}
              style={
                Array [
                  Object {
                    "color": "#FFF",
                    "fontFamily": "Inter_500Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              asd
            </Text>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFF",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "fontFamily": "Inter_500Medium",
                    },
                  ]
                }
              >
                More info
              </Text>
              <Text />
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
